src/corefx/System.Globalization.Native/icushim.cpp

```diff
@@ -170,6 +170,20 @@ bool FindLibWithMajorMinorVersion(int* majorVer, int* minorVer)
     return false;
 }
 
+// Select the version of ICU present at build time.
+bool FindLibWithBuildMajorVersion(int* majorVer)
+{
+    // ICU packaging documentation (http://userguide.icu-project.org/packaging)
+    // describes applications link against the major (e.g. libicuuc.so.54).
+    if (OpenICULibraries(U_ICU_VERSION_MAJOR_NUM, -1, -1))
+    {
+        *majorVer = U_ICU_VERSION_MAJOR_NUM;
```

Comment:
The problem with this approach is that we won't known the minor version then. On some distros, the symbol names in the ICU libraries are versioned using both major and minor versions and so we need to know both to be able to generate the symbol names.

Reply:
If the symbols are versioned at both major and minor, it would be inappropriate for those distros to include an so file that is only versioned at major.
So the open call should fail. To be safe, I can try to resolve a symbol. wdyt?